src/classes/ERR_RecordError.cls

```diff
@@ -0,0 +1,536 @@
+    }
+
+    /*******************************************************************************************************
+    * @description Takes a list of Errors, and updates the existing Errors map using the errors Record Id
+    * as a key
+    * @param errs List of Errors to be processed into the Existing Errors Map
+    */
+    @testVisible private void refreshExistingErrorsMap(List<Error__c> errs) {
+        for (Error__c err : errs) {
+            this.existingErrorsByRecordId.put (err.Record_Id__c, err);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Checks if a given DML operation has succeeded depending on the type of DML result.  Use the
+    * isSuccess() method of the result.
+    * @param Boolean true if isSuccess() was true
+    */
+    @testVisible private static boolean isDmlResultSuccess(Object dmlResult) {
+        if (dmlResult instanceof Database.SaveResult) {
+            return ((Database.SaveResult)dmlResult).isSuccess();
+        } else if (dmlResult instanceof Database.DeleteResult) {
+            return ((Database.DeleteResult)dmlResult).isSuccess();
+        } else if (dmlResult instanceof Database.UndeleteResult) {
+            return ((Database.UndeleteResult)dmlResult).isSuccess();
+        }
+        // Will only occur if a user of this class tries to use an unsupported DML Type
+        throw new RecordErrorException(System.Label.errErrorUnsupportedDMLType);
+    }
+
+    /*******************************************************************************************************
+    * @description Gets the first error from the error list if the list is not null and has at least one
+    * error present
+    * @param errs The list of Errors
+    * @return Database.Error the first error (if errors are present) or null
+    */
+    @testVisible private static Database.Error getFirstDmlError(List<Database.Error> errs) {
+        return (errs != null && errs.size() > 0) ? errs[0] : null;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates a error record associated with a specific context and record from the result of a DML
+    * operation performed through the Database class.
+    * @param result The result of the Database DML operation. It cannot be more specific than Object, because the result of
+    *               the different DML operations that the Database class can perform don't have a common parent.
+    * @param objectType The name of the type of object that caused the error.
+    * @param objId The id of the record that caused the error.
+    * @param context The context the exception was raised on.
+    * @return The error record.
+    */
+    @testVisible private static Error__c createRecordError(Object result, String objectType, Id objId, String context) {
+        Error__c error = new Error__c();
```

Comment:
I started there, and found it didn't lend itself well to this.  Keep in mind, sharing the methods will require that I make them public.  If you feel strongly about it, I will refactor for this, but it may make modifications slightly more complicated in the future.

Reply:
Very good point. They're close, but not exactly like this. A fuller refactor might have moved the `ERR_Handler` class into an instance class or a base class that we can inherit from, but I think that's a big haul for not as much gain.